tensorflow/compiler/tf2tensorrt/convert/ops/data_format_vec_permute.cc

```diff
@@ -0,0 +1,184 @@
+#include "third_party/tensorrt/NvInferRuntimeCommon.h"
+
+namespace tensorflow {
+namespace tensorrt {
+namespace convert {
+
+class ConvertDataFormatVecPermute
+    : public OpConverterBase<ConvertDataFormatVecPermute> {
+ public:
+  ConvertDataFormatVecPermute(OpConverterParams* params)
+      : OpConverterBase<ConvertDataFormatVecPermute>(params) {}
+
+  struct DataFormatVecPermuteAttributes {
+    string dst_format;
+    string src_format;
+    int spatial_dim_count;
+    int x_dim_count;
+  };
+
+  static constexpr std::array<InputArgSpec, 1> InputSpec() {
+    return {InputArgSpec::Create("x", TrtInputArg::kBoth)};
+  }
+
+  static constexpr std::array<DataType, 1> AllowedDataTypes() {
+    return {DataType::DT_INT32};
+  }
+
+  Status Validate() {
+    const auto& inputs = params_->inputs;
+    const auto& node_def = params_->node_def;
+
+    if (params_->use_implicit_batch) {
+      return errors::Unimplemented("Implicit batch mode not supported, at ",
+                                   node_def.name());
+    }
+
+    x_input_ = inputs.at(0);
+
+    // Check input rank
+    const auto x_dims = x_input_.GetTrtDims();
+    int input_rank = x_dims.nbDims;
+    if (input_rank != 1 && input_rank != 2) {
+      return errors::InvalidArgument(
+          "Input must be a vector or matrix, but got rank ", input_rank,
+          ", at ", node_def.name());
+    }
+
+    // Verify and consume node attributes
+    StatusOr<string> dst_format = GetAttrValue<string>("dst_format");
+    StatusOr<string> src_format = GetAttrValue<string>("src_format");
+    TRT_ENSURE_OK(dst_format);
+    TRT_ENSURE_OK(src_format);
+    attrs_.dst_format = *dst_format;
+    attrs_.src_format = *src_format;
+
+    // Check input dims
+    const int full_dim_count = attrs_.src_format.size();
+    const int spatial_dim_count = full_dim_count - 2;
+    if (input_rank == 1) {
+      if (x_dims.d[0] != spatial_dim_count && x_dims.d[0] != full_dim_count) {
+        return errors::InvalidArgument("1D input must be of size ",
+                                       spatial_dim_count, " or ",
+                                       full_dim_count, ", but got size ",
+                                       x_dims.d[0], ", at ", node_def.name());
+      }
+    } else if (input_rank == 2) {
+      if (x_dims.d[0] != spatial_dim_count && x_dims.d[0] != full_dim_count) {
+        return errors::InvalidArgument(
+            "First dimension of 2D input must be of size ", spatial_dim_count,
+            " or ", full_dim_count, ", but got shape (", x_dims.d[0], ", ",
+            x_dims.d[1], "), at ", node_def.name());
+      }
+      if (x_dims.d[1] != 2) {
+        return errors::InvalidArgument(
+            "Second dimension of 2D input must be of size 2, but got shape (",
+            x_dims.d[0], ", ", x_dims.d[1], "), at ", node_def.name());
+      }
+    }
+
+    // Set custom attributes
+    attrs_.spatial_dim_count = spatial_dim_count;
```

Comment:
We don't need to store this to pass to the next stage, as we can recalculate it as
attrs.src_format.size() -2

Am I right here?